optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolverFactory.java

```diff
@@ -65,24 +66,37 @@
     private static final long DEFAULT_RANDOM_SEED = 0L;
 
     private final SolverConfig solverConfig;
+    private volatile InnerScoreDirectorFactory<Solution_, ?> scoreDirectorFactory;
 
     public DefaultSolverFactory(SolverConfig solverConfig) {
         this.solverConfig = Objects.requireNonNull(solverConfig, "The solverConfig (" + solverConfig + ") cannot be null.");
     }
 
-    public InnerScoreDirectorFactory<Solution_, ?> getScoreDirectorFactory() {
-        return buildScoreDirectorFactory(solverConfig.determineEnvironmentMode());
+    public <Score_ extends Score<Score_>> InnerScoreDirectorFactory<Solution_, Score_> getScoreDirectorFactory() {
```

Comment:
This builds the score directory factory lazily. Which means the first solving requests pays the cost initialization cost.
I don't think it should be build lazily - it should be build when the SolverFactory is created.

Reply:
Nitpick: naming wise calling this get* when it actually also builds it, doesn't feel entirely right to me.